File: Java/virtual-world-simulator/src/Entities/Plants/Nightshade.java
package Entities.Plants;

import App.World;
import Common.Point;
import Entities.Entity;

import javax.swing.*;

public class Nightshade extends Plant
{
    public Nightshade(World world, Point position)
    {
        super(new ImageIcon("assets/nightshade.png"), world, position, 'N', 99, 0, SpreadChance.LOW);
    }

    public void Update()
    {
        super.Update();
        Spread(new Nightshade(this.world, new Point(-1, -1)));
    }

    public boolean Collision(Entity entity)
    {
        entity.Kill();
        Kill();
        return true;
    }
}

File: Python/virtual-world-simulator/src/Entity/Plants/Nightshade.py
import pygame
from src.Entity.Plants.Plant import *


class Nightshade(Plant):

    def __init__(self, world, position):
        temp = pygame.image.load('assets/nightshade.png')
        scaled = pygame.transform.scale(temp, (world.scale, world.scale))
        super().__init__(scaled, world, position, 'N', 99, SpreadChance.LOW)

    def update(self):
        spreadPosition = self.spread()
        if spreadPosition is not None:
            element = Nightshade(self._world, spreadPosition)
            self._world.setMapElement(spreadPosition[0], spreadPosition[1], element)
        super().update()

    def collision(self, entity):
        entity.kill()
        self.kill()
        return True
